data-prepper-plugins/s3-source/src/integrationTest/java/org/opensearch/dataprepper/plugins/source/S3ScanObjectWorkerIT.java

```diff
@@ -0,0 +1,277 @@
+    private int recordsReceived;
+    private S3ObjectGenerator s3ObjectGenerator;
+    private final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory().enable(YAMLGenerator.Feature.USE_PLATFORM_LINE_BREAKS));
+
+    private S3ObjectHandler createObjectUnderTest(final S3ObjectRequest s3ObjectRequest){
+        if(Objects.nonNull(s3ObjectRequest.getExpression()))
+            return new S3SelectObjectWorker(s3ObjectRequest);
+        else
+            return new S3ObjectWorker(s3ObjectRequest);
+    }
+
+    @BeforeEach
+    void setUp() {
+        s3Client = S3Client.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+        s3AsyncClient = S3AsyncClient.builder()
+                .region(Region.of(System.getProperty("tests.s3source.region")))
+                .build();
+        bucket = System.getProperty("tests.s3source.bucket");
+        s3ObjectGenerator = new S3ObjectGenerator(s3Client, bucket);
+        eventMetadataModifier = new EventMetadataModifier(S3SourceConfig.DEFAULT_METADATA_ROOT_KEY);
+
+        buffer = mock(Buffer.class);
+        recordsReceived = 0;
+
+        s3ObjectPluginMetrics = mock(S3ObjectPluginMetrics.class);
+        final Counter counter = mock(Counter.class);
+        final DistributionSummary distributionSummary = mock(DistributionSummary.class);
+        final Timer timer = new NoopTimer(new Meter.Id("test", Tags.empty(), null, null, Meter.Type.TIMER));
+        when(s3ObjectPluginMetrics.getS3ObjectsFailedCounter()).thenReturn(counter);
+        when(s3ObjectPluginMetrics.getS3ObjectsSucceededCounter()).thenReturn(counter);
+        when(s3ObjectPluginMetrics.getS3ObjectsFailedAccessDeniedCounter()).thenReturn(counter);
+        when(s3ObjectPluginMetrics.getS3ObjectsFailedNotFoundCounter()).thenReturn(counter);
+        when(s3ObjectPluginMetrics.getS3ObjectSizeSummary()).thenReturn(distributionSummary);
+        when(s3ObjectPluginMetrics.getS3ObjectEventsSummary()).thenReturn(distributionSummary);
+        when(s3ObjectPluginMetrics.getS3ObjectSizeProcessedSummary()).thenReturn(distributionSummary);
+        when(s3ObjectPluginMetrics.getS3ObjectReadTimer()).thenReturn(timer);
+        bucketOwnerProvider = b -> Optional.empty();
+    }
+
+    private void stubBufferWriter(final Consumer<Event> additionalEventAssertions, final String key) throws Exception {
+        doAnswer(a -> {
+            final Collection<Record<Event>> recordsCollection = a.getArgument(0);
+            assertThat(recordsCollection.size(), greaterThanOrEqualTo(1));
+            for (Record<Event> eventRecord : recordsCollection) {
+                assertThat(eventRecord, notNullValue());
+                assertThat(eventRecord.getData(), notNullValue());
+                assertThat(eventRecord.getData().get("s3/bucket", String.class), equalTo(bucket));
+                assertThat(eventRecord.getData().get("s3/key", String.class), equalTo(key));
```

Comment:
Can you also assert the actual event data here?

Reply:
Thanks for your review Krishna, assert event data already validating in the RecordsGenerator implementations. so i think we no need to verify explicitly here. Please suggest.

data-prepper-plugins/s3-source/src/integrationTest/java/org/opensearch/dataprepper/plugins/source/ParquetRecordsGenerator.java
Line No: 45
Method Name:  assertEventIsCorrect(final Event event)